Ginger/Ginger/UserControlsLib/ucGridView/ucGrid.xaml.cs

```diff
@@ -1716,6 +1719,41 @@ private void AddToolbarTool(object userControl, string toolTip = "", RoutedEvent
         }
 
 
+        public void AddComboBoxToolbarTool(string lableContent,Type enumType, SelectionChangedEventHandler view_SelectionChanged)
+        {
+            ComboBox comboBox = new ComboBox();
+            comboBox.Style = this.FindResource("$FlatInputComboBoxStyle") as Style;
+
+            comboBox.Width = 100;
+            List<GingerCore.General.ComboEnumItem> itemsList = GingerCore.General.GetEnumValuesForCombo(enumType);
+            List<GingerCore.General.ComboEnumItem> itemsSourceList = new List<GingerCore.General.ComboEnumItem>();
+            itemsSourceList = itemsList;
+            comboBox.ItemsSource = itemsSourceList;
+            comboBox.SelectedIndex = 0;
+            comboBox.SelectionChanged += view_SelectionChanged;
+            comboBox.Style = (Style)TryFindResource("@ToolBarComboBoxStyle");
+
+            Label label = new Label();
+            label.Content = lableContent;
+
+            toolbar.Items.Remove(lblSearch);
```

Comment:
i know it was here before for "AddToolbarTool", but can we enter the new ComboBoxTool in specific index instead of removing all and re-adding it, currently it is ugly solution.

Reply:
did you tried with index approch?